CODING_STYLE.md

```diff
@@ -0,0 +1,199 @@
+# C++ Ethereum Coding Style
+
+## Code Formatting
+
+Use clang-format tool to format your changes, see [CONTRIBUTING](CONTRIBUTING.md) for details.
+
+
+## Namespaces
+
+a. No "using namespace" declarations in header files.
+b. All symbols should be declared in a namespace except for final applications.
+c. Preprocessor symbols should be prefixed with the namespace in all-caps and an underscore.
+
+    // WRONG:
+    #include <cassert>
+    using namespace std;
+    tuple<float, float> meanAndSigma(vector<float> const& _v);
+
+    // CORRECT:
+    #include <cassert>
+    std::tuple<float, float> meanAndSigma(std::vector<float> const& _v);
+
+
+## Preprocessor
+
+1. File comment is always at top, and includes:
+   - Copyright.
+   - License.
+   
+2. Never use #ifdef/#define/#endif file guards. Prefer #pragma once as first line below file comment.
+3. Prefer static const variable to value macros.
+4. Prefer inline constexpr functions to function macros.
+
+
+## Capitalization
+
+GOLDEN RULE: Preprocessor: ALL_CAPS; C++: camelCase.
+
+1. Use camelCase for splitting words in names, except where obviously extending STL/boost functionality in which case follow those naming conventions.
+2. The following entities' first alpha is upper case:
+   - Type names.
+   - Template parameters.
+   - Enum members.
+   - static const variables that form an external API.
+3. All preprocessor symbols (macros, macro arguments) in full uppercase with underscore word separation.
+
+
+All other entities' first alpha is lower case.
+
+
+## Variable prefixes
+
+1. Leading underscore "_" to parameter names.
+   - Exception: "o_parameterName" when it is used exclusively for output. See 6(f).
+   - Exception: "io_parameterName" when it is used for both input and output. See 6(f).
+2. Leading "c_" to const variables (unless part of an external API).
+3. Leading "g_" to global (non-const) variables.
+4. Leading "s_" to static (non-const, non-global) variables.
+
+
+
+## Error reporting
+
+Prefer exception to bool/int return type.
+
+
+## Declarations
+
+1. {Typename} + {qualifiers} + {name}. (**TODO**: Against [NL.26](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl26-use-conventional-const-notation))
+2. Only one per line.
+3. Favour declarations close to use; don't habitually declare at top of scope ala C.
+4. Always pass non-trivial parameters with a const& suffix.
+5. If a function returns multiple values, use std::tuple (std::pair acceptable).
+   Prefer not using */& arguments, except where efficiency requires.
+   (**TODO**: Compare with [C++ Core Guidelines F.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f21-to-return-multiple-out-values-prefer-returning-a-tuple-or-struct) )
```

Comment:
I think we agree overall with Core Guidelines here, so maybe put just `See also [C++ Core Guidelines F.21]`